allow to add a personal note to a share

schiessle opened the pull request:
Allow users to add a personal note to a share which will then be send to the recipient (in case of a user, group or mail share) or displayed at the public link page (in case of a link or mail share)

- [x] back-end to set/get a note to a share
- [x] sending notes by mail implemented for user, group and mail shares
- [x] UI to add a note to a share (user, group, mail)
- [x] UI to add note to public links
- [x] display note on public link page (maybe it could look nice, but it works :wink: )
- [x] fix/update unit tests

For the UI to add a note to a share I would suggest to add another menu option to the "..." menu for user, group and mail shares called "Add note" if the user clicks on it we show a input field to add the note. For public links I would add a similar option under the check-boxes.

After the user set a note it is send to the server over the already existing OCS API, same as we already use to update permission, expire date, etc. Example call:

````
curl -X PUT http://admin:admin@localhost/server/ocs/v2.php/apps/files_sharing/api/v1/shares/1 -d note="hello world2!" -H OCS-APIRequest:true
````

schiessle committed:
allow to add a personal note to a share

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle commented:
@skjnldsv maybe you have some time to look at the UI part?

nickvergessen commented on apps/sharebymail/lib/ShareByMailProvider.php:
```diff
@@ -506,6 +506,61 @@ protected function sendPassword(IShare $share, $password) {
+		$initiatorDisplayName = ($initiatorUser instanceof IUser) ? $initiatorUser->getDisplayName() : $initiator;
+		$initiatorEmailAddress = ($initiatorUser instanceof IUser) ? $initiatorUser->getEMailAddress() : null;
+
+		$plainBodyPart = $this->l->t("%s shared »%s« with you and want to add:\n", [$initiatorDisplayName, $filename]);
```
* Single quotes
* Please use counted placeholders: `%$1s` and `%$2s`, otherwise Right-to-Left languages have problems.

schiessle replied:
done

schiessle committed:
improve mail template and escape html code

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle committed:
add note to share object when reading it from the db

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

nickvergessen commented:
If okay, I would only allow this for mail/public shares.
Internal shares should use comments?

nickvergessen commented on apps/sharebymail/lib/ShareByMailProvider.php:
```diff
@@ -506,6 +506,54 @@ protected function sendPassword(IShare $share, $password) {
+		$initiatorDisplayName = ($initiatorUser instanceof IUser) ? $initiatorUser->getDisplayName() : $initiator;
+		$initiatorEmailAddress = ($initiatorUser instanceof IUser) ? $initiatorUser->getEMailAddress() : null;
+
+		$plainHeading = $this->l->t('%1s shared »%2s« with you and want to add:', [$initiatorDisplayName, $filename]);
```
Missing the `$` between the number and s: `%1$s` and `%2$s`

nickvergessen commented on lib/private/Share20/DefaultShareProvider.php:
```diff
@@ -1227,4 +1263,91 @@ protected function filterSharesOfUser(array $shares) {
+			// The "From" contains the sharers name
+			$instanceName = $this->defaults->getName();
+			$senderName = $this->l->t(
+				'%s via %s',
```
`%1$s`

nickvergessen commented on lib/private/Share20/DefaultShareProvider.php:
```diff
@@ -1227,4 +1263,91 @@ protected function filterSharesOfUser(array $shares) {
+			$initiatorDisplayName = ($initiatorUser instanceof IUser) ? $initiatorUser->getDisplayName() : $initiator;
+			$initiatorEmailAddress = ($initiatorUser instanceof IUser) ? $initiatorUser->getEMailAddress() : null;
+
+			$plainHeading = $this->l->t('%1s shared »%2s« with you and want to add:', [$initiatorDisplayName, $filename]);
```
`%1$s`

schiessle committed:
show note on public link page

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle commented:
> If okay, I would only allow this for mail/public shares.
> Internal shares should use comments?

No, one of the important use case is to avoid the typical work flow: share a document with a colleague, go to the mail client and write a mail "hey, I just have shared document X with you. Can you please review it so that we can do XYZ". So it is important that it also works for internal shares.

Btw, this reminds me that I should add a link to the internal share to the mail. Is there a easy way to generate it?

schiessle committed:
fix mail templates

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle committed:
add direct link to the file for user/group shares

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle committed:
add direct link to the share

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle committed:
add a nice looking button as direct link to the sharre

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle committed:
fix unit tests

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

nickvergessen commented on core/Migrations/Version14000Date20180712153140.php:
```diff
@@ -0,0 +1,45 @@
+
+		$table = $schema->getTable('share');
+		$table->addColumn('note', 'text', [
+			'notnull' => true,
```
any reason for this?
Just received the error when testing:
```
An exception occurred while executing 'INSERT INTO `oc_share` (`share_type`, `item_type`, `item_source`, `file_source`, `share_with`, `uid_owner`, `uid_initiator`, `permissions`, `token`, `password`, `stime`, `file_target`) VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [4, "file", 3, 3, "test@****.com", "admin", "admin", 1, "dLzxBkWpFcjZnFN", null, 1531473009, ""]: SQLSTATE[HY000]: General error: 1364 Field 'note' doesn't have a default value
```

schiessle committed:
allow 'note' to be null

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

nickvergessen commented on apps/files_sharing/templates/public.php:
```diff
@@ -31,6 +31,11 @@
 	<div id="files-public-content">
+		<?php if (isset($_['note']) && $_['note'] !== '') : ?>
+		<div id="note">
+			<?php p($l->t('Note: ')); p($_['note']); ?>
```
space should not be part of the translation, or you insert the note with a placeholder

nickvergessen commented on apps/sharebymail/lib/ShareByMailProvider.php:
```diff
@@ -506,6 +506,61 @@ protected function sendPassword(IShare $share, $password) {
+		$initiatorEmailAddress = ($initiatorUser instanceof IUser) ? $initiatorUser->getEMailAddress() : null;
+
+		$plainHeading = $this->l->t('%1$s shared »%2$s« with you and want to add:', [$initiatorDisplayName, $filename]);
+		$htmlHeading = $this->l->t('%1$s shared »%2$s« with you and want to add:', [$initiatorDisplayName, $filename]);
```
*wants (3rd person singular -  he/she/it, das s muss mit :) )

nickvergessen replied:
I think in the HTML email the trailing `:` looks strange

nickvergessen commented:
Currently it always sends a new email when the note is updated, is that intentional?
Could otherwise also check if the note was already set and only send the email when it was empty before

schiessle commented:
> Currently it always sends a new email when the note is updated, is that intentional?

Well, good question. I thought that it might make sense to send the note again if the user changes it. If you replace the note with another one you would expect that the user receives it, right?

If we send it only once we would also need to block editing once a note was set, but this could confuse people if they can't fix a typo, extend the note, etc.

schiessle committed:
some small template fixes

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle committed:
explicitely mention that the note column can be null

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle committed:
fix unit tests

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle committed:
return the as well note if we fetch a share

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle committed:
unit tests updated

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

skjnldsv committed:
UI set share note!

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
![capture d ecran_2018-07-14_12-03-49](https://user-images.githubusercontent.com/14975046/42723432-fee21c2e-875d-11e8-8248-576aa3c3af51.png)

schiessle commented:
Would it be possible to grow the open "..." menu and put the input field in the menu? Putting it below the share link looks confusing. Especially if there are more then one share and maybe even a public link.

When I try to save a note I get "An error occurred. Unable to save the note". You send a PUT request to " http://localhost/server/index.php/apps/files/", the right URL would be `http://localhost/server/ocs/v2.php/apps/files_sharing/api/v1/shares/<shareid>`, see curl example at the top.

I also couldn't find a way to set a not for a public link share.

schiessle commented:
@skjnldsv afaik the design was from a stage of discussion where we considered haveing only one comment per file (for all shares). But as we now want to have one comment per share I think it makes more sense to display the input filed more next to the share like we do it with the password field for mail shares:

![image](https://user-images.githubusercontent.com/1589737/42749313-92202f5e-88e3-11e8-832f-4cd69dab33bb.png)


For the link shares I would just add another checkbox and behave also the same as the "password" field for link shares behave.

schiessle commented:
Does anyone know why this tests fail: https://drone.nextcloud.com/nextcloud/server/8944/107

When I saw it the first time I fixed it here: https://github.com/nextcloud/server/pull/10218/commits/a0ac9b3c00492a8bfd255317fb5e74591c80464c

Now all tests pass locally but it seems like CI ignores this commit and :confused:

jospoortvliet commented:
> > If okay, I would only allow this for mail/public shares.
> > Internal shares should use comments?
> 
> No, one of the important use case is to avoid the typical work flow: share a document with a colleague, go to the mail client and write a mail "hey, I just have shared document X with you. Can you please review it so that we can do XYZ". So it is important that it also works for internal shares.

Wouldn't it make sense to 'simply' make this the first comment on the file? Sorry, just thinking out loud...

schiessle commented:
> Wouldn't it make sense to 'simply' make this the first comment on the file? Sorry, just thinking out loud...

No, because it should be personal 1:1 comment, only seen by the recipient not "randomly" the first comment seen by everyone.

MorrisJobke commented:
> The current implementation has one drawback: Notes for internal user/group shares might never reach the user if no email address is configured. Could be nice to display them somehow in the file listing, people might still miss it if they use desktop/mobile clients only but at least it is visible "somehow". But don't know how hard it would be to add it... @skjnldsv (would also put it at the end of the todo list)

It's kind of weird. Shouldn't it be just a comment?

jancborchardt commented:
I mean, we had a discussion about this, no? And we discussed the proper way to do this is:

- Have a sidebar in shares to display comments
- Be able to toggle certain file comments to be publicly visible

Adding yet another different way to add text to a file (remember we will already have the Comments vs Talk chat issue) will just be a mess.

skjnldsv committed:
Popovermenu migration

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Autoresize + autoloader

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

schiessle commented:
There is still something wrong how you build the URL to send the request, the PUT request is send to:

`http://localhost/server/ocs/v2.php/apps/files_sharing/api/v1/shares/5`

while this would the the correct URL on my test system:

`http://localhost/server/index.php/ocs/v2.php/apps/files_sharing/api/v1/shares/5`

Beside that, directly after the request was send another request is send to `http://localhost/server/index.php/apps/files/` with the note

schiessle committed:
fix post url

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle commented:
I changed now the way the URL is created and it works now for me :slightly_smiling_face: 

The popover looks also nice :+1: 

Now we only need to add it for link shares.

schiessle committed:
fix unit tests

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle commented:
@jancborchardt @MorrisJobke 

This is not about comments but about a personal note. What people does today is:
1. sharing a file
2. go to their email client and write a mail along the line "I just shared with you X, can you please do Y and check the stupid mistake from Z"

This is  a personal note (not a general comment, that's why we also choosed a different name) which should only be visible to the intended person. The feature here is to prevent the switch between your Nextcloud where you created the share and the switch to you email client to explain to people what and why you shared X with them. Beside that please look at the linked discussion in the other issue. All this was already discussed in great detail.

schiessle commented:
@skjnldsv I identified some small issues, maybe you can have a look:

1. JS tests are broken
2. The Notes option should be only available for mail, user, group shares and for public links. Not for federated user (`\OCP\Share::SHARE_TYPE_REMOTE_GROUP`) or group shares (`\OCP\Share::SHARE_TYPE_REMOTE_GROUP`) for now.
3. Already know, but add it for a complete list: Add the option for link shares, in doubt I would just copy the "password" checkbox for now. So that we have at least something feature complete to test.

skjnldsv committed:
Public share

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

MorrisJobke commented:
> http://localhost/server/index.php/ocs/v2.php/apps/files_sharing/api/v1/shares/5

Woot? This looks so wrong. `v2.php` is an independent endpoint that is not proxied through index.php

nickvergessen commented:
> Woot? This looks so wrong. v2.php is an independent endpoint that is not proxied through index.php

See https://github.com/nextcloud/server/pull/10218/commits/b545af94ffedd1682bc121d383c7914548069983 :wink:

schiessle commented:
> Woot? This looks so wrong. v2.php is an independent endpoint that is not proxied through index.php

right, was my mistake and is fixed now... It was a different problem.

MorrisJobke commented:
> This is not about comments but about a personal notes.

Is there any way to include the comment right into the share link email otherwise it is kind of weird, because the recipient still needs to relate the comment somehow to the link and if a person sends two or three shares and add notes it is a total of 6 mails with links and comments mixed. Just some ideas, but it seems you said, that this is okay.

Is it also shown in the public share page? Because just dumping the text ontop of a share looks a bit ugly.

schiessle commented:
> Is it also shown in the public share page? Because just dumping the text ontop of a share looks a bit ugly.

Yes, for public links it will be also show on the public share page

jancborchardt reviewed (changes requested):
- [x] When clicking "Set share note", the textarea should be focused
- [x] For user shares, the "Set expiration date" input field should be below the text in a new row, as it was before
- [x] For Share link, the Share note textarea doesn’t seem to do autosize (for user shares it does)

Otherwise good! :)

skjnldsv committed:
Public fixes

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

jancborchardt reviewed (approved):
Looks great now! :) :+1:

schiessle commented:
hm, the share sidebar is now empty for me. The browser log says:

````
TypeError: OC.Share[className] is not a constructor[Learn More] sharedialogview.js:121:8
	initialize http://localhost/server/core/js/sharedialogview.js:121:8
	Backbone.View http://localhost/server/core/vendor/backbone/backbone.js:1192:5
	child http://localhost/server/core/vendor/backbone/backbone.js:1852:34
	render http://localhost/server/apps/files_sharing/js/sharetabview.js:68:20
	setFileInfo http://localhost/server/apps/files/js/detailtabview.js:80:5
	selectTab http://localhost/server/apps/files/js/detailsview.js:195:4
	showDetailsView http://localhost/server/apps/files/js/filelist.js:519:5
	actionHandler http://localhost/server/apps/files_sharing/js/share.js:171:7
	action http://localhost/server/apps/files/js/fileactions.js:147:6
	_renderInlineAction/< http://localhost/server/apps/files/js/fileactions.js:442:6
	dispatch jquery.js:4434:14
	add/r.handle jquery.js:4121:5
````

the nextcloud.log also has a lot of errors every time I reload the page, not all might be relevant:

````
{"reqId":"AODYatfZXoPufDWukxzh","level":3,"time":"2018-07-19T18:59:02+00:00","remoteAddr":"127.0.0.1","user":"admin","app":"core","method":"GET","url":"\/server\/index.php\/apps\/files\/?dir=\/","message":{"Exception":"Exception","Message":"Could not find category \"_$!<Favorite>!$_\"","Code":0,"Trace":[{"file":"\/home\/schiesbn\/Repos\/nextcloud\/server\/lib\/private\/Tags.php","line":628,"function":"getIdsForTag","class":"OC\\Tags","type":"->","args":["_$!<Favorite>!$_"]},{"file":"\/home\/schiesbn\/Repos\/nextcloud\/server\/apps\/files\/lib\/Activity\/Helper.php","line":51,"function":"getFavorites","class":"OC\\Tags","type":"->","args":[]},{"file":"\/home\/schiesbn\/Repos\/nextcloud\/server\/apps\/files\/lib\/Controller\/ViewController.php","line":163,"function":"getFavoriteFilePaths","class":"OCA\\Files\\Activity\\Helper","type":"->","args":["admin"]},{"file":"\/home\/schiesbn\/Repos\/nextcloud\/server\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":166,"function":"index","class":"OCA\\Files\\Controller\\ViewController","type":"->","args":["\/","",null,false]},{"file":"\/home\/schiesbn\/Repos\/nextcloud\/server\/lib\/private\/AppFramework\/Http\/Dispatcher.php","line":99,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Files\\Controller\\ViewController"},"index"]},{"file":"\/home\/schiesbn\/Repos\/nextcloud\/server\/lib\/private\/AppFramework\/App.php","line":118,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Files\\Controller\\ViewController"},"index"]},{"file":"\/home\/schiesbn\/Repos\/nextcloud\/server\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["ViewController","index",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"files.view.index"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"files.view.index"}]},{"file":"\/home\/schiesbn\/Repos\/nextcloud\/server\/lib\/private\/Route\/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"files.view.index"}]},{"file":"\/home\/schiesbn\/Repos\/nextcloud\/server\/lib\/base.php","line":989,"function":"match","class":"OC\\Route\\Router","type":"->","args":["\/apps\/files\/"]},{"file":"\/home\/schiesbn\/Repos\/nextcloud\/server\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/home\/schiesbn\/Repos\/nextcloud\/server\/lib\/private\/Tags.php","Line":282,"CustomMessage":"OC\\Tags::getFavorites"},"userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:61.0) Gecko\/20100101 Firefox\/61.0","version":"14.0.0.11"}
{"reqId":"AODYatfZXoPufDWukxzh","level":3,"time":"2018-07-19T18:59:02+00:00","remoteAddr":"127.0.0.1","user":"admin","app":"cssresourceloader","method":"GET","url":"\/server\/index.php\/apps\/files\/?dir=\/","message":"Could not find resource core\/css\/share.css to load","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:61.0) Gecko\/20100101 Firefox\/61.0","version":"14.0.0.11"}
````

schiessle commented:
JS tests still fails: https://drone.nextcloud.com/nextcloud/server/9069/247

schiessle commented:
If I skip your last commit it works, seems there you broke something.
The new share link dialog looks nice! :slightly_smiling_face: 

Beside that, please don't forget the points mentioned here: https://github.com/nextcloud/server/pull/10218#issuecomment-406187240

schiessle commented:
@skjnldsv 
If I skip your last commit it works, seems there you broke something.

The new public share dialog looks nice! :slightly_smiling_face: 

Beside that, please don't forget the points mentioned here: https://github.com/nextcloud/server/pull/10218#issuecomment-406187240

schiessle committed:
move the note up, so that it is also visible in the file drop view

Signed-off-by: Bjoern Schiessle <bjoern@schiessle.org>

schiessle commented:
@jancborchardt did you had a look at the note on the public page? Right now it is just a text `Note: <note>` at the top:

![image](https://user-images.githubusercontent.com/1589737/42968877-dceb8394-8ba4-11e8-9b6f-b2f1b3b874b7.png)

![image](https://user-images.githubusercontent.com/1589737/42968895-ea532cc6-8ba4-11e8-9baa-35dac2f5b80f.png)

![image](https://user-images.githubusercontent.com/1589737/42968994-1c70e0e0-8ba5-11e8-8bef-1dfd796f2e0f.png)

Maybe this could look a bit nicer? Put it in a box (border, background color,...)? Font changes? A icon or something like this next to the note... I don't really know but as of now it looks not that nice. Any ideas?

skjnldsv commented:
@schiessle what last commit?
Everything works for me? Have you cleared your cache? :)

jancborchardt commented:
@schiessle my proposal would be to show an icon-comment in the header, and on clicking that showing a popover with the note in it, saying:

> **Note from Alice**
> Hi Bob, here’s the vacation pictures.
> You can upload your own in the folder too.

This is relatively simple and scalable to larger notes also, and should also work on mobile (if we put the filename and "shared by" _above_/_below_ instead of left/right next to each other.

MorrisJobke commented:
> @jancborchardt did you had a look at the note on the public page? Right now it is just a text Note: <note> at the top:

Isn't the idea to first ask the design team for ways how to implement this and then implement it 🤔 

Because this one looks not really good. it looks like just duck-taped into it without anybody thinking about it.

schiessle commented:
> Isn't the idea to first ask the design team for ways how to implement this and then implement it 

The first mookups which looks quite similar as it looks now exists already for quite some time. The design team could have had a look at it already earlier :wink: 

Although it is not so unusual that you first get something working and then polish it... That's why I asked Jan here if he actually looked at it.

MorrisJobke commented:
> The first mookups which looks quite similar as it looks now exists already for quite some time. The design team could have had a look at it already earlier 😉

But it wasn't mentioned in that ticket ;)



> Although it is not so unusual that you first get something working and then polish it... That's why I asked Jan here if he actually looked at it.

The question from me here was because one should first think about the rough idea, then implement it and then polish it and not put it just randomly somewhere.

Anyways. Let's get this in and think about how to improve it, because this one is not really something we should ship. Jan already had an idea how to improve this.

schiessle commented:
> The question from me here was because one should first think about the rough idea, then implement it and then polish it and not put it just randomly somewhere.

Com'on Morris. This was a lot of planing, discussing about the idea and how to do it and you know it, or at least you know where to look it up. That's not a midnight hack which come out of nowhere.

skjnldsv committed:
No remote note and various fixes

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
@schiessle good now?

schiessle commented:
@skjnldsv great! Everything works now... And the share dialog looks really nice. I like it a lot, great work! :+1:

juliusknorr committed:
Don't use shorthand syntax for functions

Signed-off-by: Julius Härtl <jus@bitgrid.net>

MorrisJobke commented:
Tested and works 👍 No @skjnldsv and @jancborchardt just need to polish a bit the public share page and this is good to go in.

skjnldsv committed:
Tests fixes and design bump

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

jancborchardt commented:
Here's what we planned for the public page:
• Move "shared by" info below document title
• Show person's avatar with red dot and short pulsing initially (like Notifications app)
• Tapping will open a popover with a header saying "Note from username" and the note below

![](https://i.imgur.com/YtZHd0D.jpg)

rullzer reviewed (approved):
Works here!
Lets do this

MorrisJobke commented:
Acceptance tests are failing with this:

```
    And I share the link for "Editable shared folder"                                 # FilesAppContext::iShareTheLinkFor()
      Share link checkbox in the details view in Files app could not be found after 50 seconds (NoSuchElementException)
```

skjnldsv committed:
Design structure fixes

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv committed:
Design fixes

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

danxuliu committed:
Adjust acceptance tests to the new share link UI

The share link UI no longer uses its own layout below the other shares;
now it is shown as a share row with a menu for the actions (except
enabling it, which is shown in the row itself), just like the other
shares.

The share link is no longer shown, either; now the link is got by
clicking on a "Copy URL" menu item, which copies the link to the
clipboard. As the clipboard is not accessible from the acceptance tests
the URL is now extracted from the attributes of that menu item (although
the menu item is clicked anyway to mimic the user behaviour).

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

danxuliu commented:
Acceptance tests fixed; the test _Viewing a favorite file in its folder does not prevent opening the details view in "All files" section_ fails not due to changes in this pull request, but due to #9982

skjnldsv committed:
Comply @since requirements

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented:
Failure unrelated for files : `sh: 1: kill: No such process`

I'll handle the user acceptance failure in another pr since it's on master!
